src/rogue/interfaces/stream/Slave.cpp

```diff
@@ -70,11 +73,13 @@ void ris::Slave::acceptFrame ( ris::FramePtr frame ) {
       log_->log(100,"Got Size=%i, Data:",frame->getPayload());
       sprintf(buffer,"     ");
 
-      for (x=0; (x < debug_ && x < frame->getPayload()); x++) {
-         frame->read(&val,x,1);
+      count = 0;
+      for (it = frame->begin(); (count < debug_) && (it != frame->end()); ++it) {
+         count++;
+         val = *it;
 
          snprintf(buffer + strlen(buffer),1000-strlen(buffer)," 0x%.2x",val);
-         if (( (x+1) % 8 ) == 0) {
+         if (( (count+1) % 8 ) == 0) {
             log_->log(100,buffer);
```

Comment:
```log_->log(100,......);```
Isn't there a header constant that you should use instead of hard coding to 100?

Reply:
That should probably just be critical.